http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpAppDAO.scala

```diff
@@ -25,14 +34,22 @@ class HttpAppDAO[F[_]: Async](val kubernetesDnsCache: KubernetesDnsCache[F], cli
           .status(
             Request[F](
               method = Method.GET,
-              uri = Uri.unsafeFromString(serviceUrl)
+              uri = Uri.unsafeFromString(serviceUrl),
+              headers = Headers(Header.Raw(CIString("X-Request-ID"), traceId.asString))
             )
           )
           .map(status => status.code < 400) // consider redirect also as success
+          .handleErrorWith(t =>
+            logger.error(Map("traceId" -> traceId.asString), t)("Fail to check if app is up").as(false)
```

Comment:
this is the main fix. essentially if we see an exception, we'll translate that to `false`, meaning the app is not up yet (instead of throwing exception and cause polling to stop)